orchagent/pbhorch.cpp

```diff
@@ -0,0 +1,1395 @@
+    {
+        SWSS_LOG_ERROR("Failed to remove PBH hash(%s) from SAI", hObj.key.c_str());
+        return false;
+    }
+
+    if (!this->pbhHlpr.removePbhHash(hObj.key))
+    {
+        SWSS_LOG_ERROR("Failed to remove PBH hash(%s) from internal cache", hObj.key.c_str());
+        return false;
+    }
+
+    if (!this->pbhHlpr.removeDependencies(hObj))
+    {
+        SWSS_LOG_ERROR("Failed to remove PBH hash(%s) dependencies", hObj.key.c_str());
+        return false;
+    }
+
+    SWSS_LOG_NOTICE("Removed PBH hash(%s) from SAI", hObj.key.c_str());
+
+    return true;
+}
+
+void PbhOrch::deployPbhHashSetupTasks()
+{
+    SWSS_LOG_ENTER();
+
+    auto &map = this->pbhHlpr.hashTask.pendingSetupMap;
+    auto it = map.begin();
+
+    while (it != map.end())
+    {
+        auto &key = it->first;
+        auto &hash = it->second;
+
+        if (!this->pbhHlpr.validateDependencies(hash))
+        {
+            SWSS_LOG_NOTICE("Unable to create PBH hash(%s): object has missing dependencies: adding a retry", key.c_str());
+            it++;
+            continue;
+        }
+
+        PbhHash hObj;
+
+        if (!this->pbhHlpr.getPbhHash(hObj, key))
+        {
+            if (!this->createPbhHash(hash))
+            {
+                SWSS_LOG_ERROR("Failed to create PBH hash(%s): ASIC and CONFIG DB are diverged", key.c_str());
+            }
+        }
+        else
+        {
+            if (!this->updatePbhHash(hash))
+            {
+                SWSS_LOG_ERROR("Failed to update PBH hash(%s): ASIC and CONFIG DB are diverged", key.c_str());
+            }
+        }
+
+        it = map.erase(it);
+    }
+}
+
+void PbhOrch::deployPbhHashRemoveTasks()
+{
+    SWSS_LOG_ENTER();
+
+    auto &map = this->pbhHlpr.hashTask.pendingRemoveMap;
+    auto it = map.begin();
+
+    while (it != map.end())
+    {
+        auto &key = it->first;
+        auto &hash = it->second;
+
+        PbhHash hObj;
+
+        if (!this->pbhHlpr.getPbhHash(hObj, key))
+        {
+            SWSS_LOG_ERROR("Failed to remove PBH hash(%s): object doesn't exist", key.c_str());
+            it = map.erase(it);
+            continue;
+        }
+
+        if (this->pbhHlpr.hasDependencies(hObj))
+        {
+            SWSS_LOG_NOTICE("Unable to remove PBH hash(%s): object has dependencies: adding a retry", key.c_str());
+            it++;
+            continue;
+        }
+
+        if (!this->removePbhHash(hash))
+        {
+            SWSS_LOG_ERROR("Failed to remove PBH hash(%s): ASIC and CONFIG DB are diverged", key.c_str());
+            it = map.erase(it);
+            continue;
+        }
+
+        it = map.erase(it);
+    }
+}
+
+// PBH hash field -----------------------------------------------------------------------------------------------------
+
+bool PbhOrch::createPbhHashField(const PbhHashField &hashField)
+{
+    SWSS_LOG_ENTER();
+
+    PbhHashField hfObj;
+
+    if (this->pbhHlpr.getPbhHashField(hfObj, hashField.key))
```

Comment:
You are checking this in caller(deployPbhHashFieldSetupTasks) already, any value in adding this check in both places?

Reply:
Similar comment applies for other fns in this file as well, eg: updatePbhHashField and others

Reply:
@anish-n this is an additional check just in case if the method will be called from another API. For now we can remove it as you mentioned.